diff --git a/sources/hyperhdr/CMakeLists.txt b/sources/hyperhdr/CMakeLists.txt
index cd5d5bc..edff19f 100644
--- a/sources/hyperhdr/CMakeLists.txt
+++ b/sources/hyperhdr/CMakeLists.txt
@@ -181,14 +181,6 @@ if(WIN32)
 		endif()
 endif()		
 
-if (APPLE)
-	DeployApple("hyperhdr")
-elseif (NOT WIN32) # Unix
-	DeployUnix("hyperhdr")
-elseif(WIN32) # Windows
-	DeployWindows("hyperhdr")
-endif ()
-
 if(USE_PRECOMPILED_HEADERS AND COMMAND target_precompile_headers)
     target_precompile_headers(hyperhdr REUSE_FROM precompiled_hyperhdr_headers)
 endif()
