diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ac3be4..874a4a7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -638,7 +638,3 @@ else()
 	set_target_properties(resources uninstall PROPERTIES FOLDER ExternalLibsTargets)
 endif()
 
-if (NOT ENABLE_BONJOUR)	
-	message( FATAL_ERROR "The network discovery service (ENABLE_BONJOUR) is an important feature of HyperHDR. If you know what you are doing, remove this safe-break.")
-endif()
-
