diff --git a/src/plugins/plugins.pro b/src/plugins/plugins.pro
index c07b02b1..557a1198 100644
--- a/src/plugins/plugins.pro
+++ b/src/plugins/plugins.pro
@@ -10,7 +10,6 @@ SUBDIRS   = \
     texteditor \
     cppeditor \
     bineditor \
-    boot2qt \
     diffeditor \
     imageviewer \
     bookmarks \
@@ -42,7 +41,6 @@ SUBDIRS   = \
     tasklist \
     qmljstools \
     macros \
-    remotelinux \
     android \
     valgrind \
     todo \
