diff --git a/Platform/Linux-gdc.cmake b/Platform/Linux-gdc.cmake
index 0f31c26..c866eb8 100644
--- a/Platform/Linux-gdc.cmake
+++ b/Platform/Linux-gdc.cmake
@@ -32,7 +32,7 @@ endif()
 set(CMAKE_D_FLAGS_INIT "")
 set(CMAKE_D_FLAGS_DEBUG_INIT "-g ${DDOC_FLAGS}")
 set(CMAKE_D_FLAGS_MINSIZEREL_INIT "-Os ${DDOC_FLAGS}")
-set(CMAKE_D_FLAGS_RELEASE_INIT "-O3 -fomit-frame-pointer -fweb -frelease -finline-functions ${DDOC_FLAGS}")
+set(CMAKE_D_FLAGS_RELEASE_INIT "-O2 -fomit-frame-pointer -fweb -frelease -finline-functions ${DDOC_FLAGS}")
 set(CMAKE_D_FLAGS_RELWITHDEBINFO_INIT "-O2 -g ${DDOC_FLAGS}")
 # set(CMAKE_D_CREATE_PREPROCESSED_SOURCE "<CMAKE_D_COMPILER> <FLAGS> -E <SOURCE> > <PREPROCESSED_SOURCE>")
 set(CMAKE_D_CREATE_ASSEMBLY_SOURCE "<CMAKE_D_COMPILER> <FLAGS> -S <SOURCE> -o <ASSEMBLY_SOURCE>")
diff --git a/Platform/Windows-gdc.cmake b/Platform/Windows-gdc.cmake
index a476fe3..45a3042 100644
--- a/Platform/Windows-gdc.cmake
+++ b/Platform/Windows-gdc.cmake
@@ -28,7 +28,7 @@ endif()
 set(CMAKE_D_FLAGS_INIT "-fversion=Win -fversion=${CMAKE_BUILD_TYPE}Build ${DSTDLIB_FLAGS}")
 set(CMAKE_D_FLAGS_DEBUG_INIT "-g ${DDOC_FLAGS}")
 set(CMAKE_D_FLAGS_MINSIZEREL_INIT "-Os ${DDOC_FLAGS}")
-set(CMAKE_D_FLAGS_RELEASE_INIT "-O3 -fomit-frame-pointer -fweb -frelease -finline-functions ${DDOC_FLAGS}")
+set(CMAKE_D_FLAGS_RELEASE_INIT "-O2 -fomit-frame-pointer -fweb -frelease -finline-functions ${DDOC_FLAGS}")
 set(CMAKE_D_FLAGS_RELWITHDEBINFO_INIT "-O2 -g ${DDOC_FLAGS}")
 # set(CMAKE_D_CREATE_PREPROCESSED_SOURCE "<CMAKE_D_COMPILER> <FLAGS> -E <SOURCE> > <PREPROCESSED_SOURCE>")
 set(CMAKE_D_CREATE_ASSEMBLY_SOURCE "<CMAKE_D_COMPILER> <FLAGS> -S <SOURCE> -o <ASSEMBLY_SOURCE>")
