diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8301a1..7895d51 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ set(MOD_SRCS
 	UseDDoc.cmake
 	UseDDeps.cmake
 	UseDub.cmake
-	dependencies.cmake
+	DLangDependencies.cmake
 	UseDUnittest.cmake
 	FindPhobos.cmake
 	CMakePlatformId.di.in
diff --git a/dependencies.cmake b/DLangDependencies.cmake
similarity index 100%
rename from dependencies.cmake
rename to DLangDependencies.cmake
diff --git a/UseDDeps.cmake b/UseDDeps.cmake
index 41284f5..3c12b0e 100644
--- a/UseDDeps.cmake
+++ b/UseDDeps.cmake
@@ -39,7 +39,7 @@ macro(add_executable_with_dependencies _target)
       -D "include_directories:STRING=${INCLUDES}"
       -D "source_file:STRING=${source_file}"
       -D "dependency_file:STRING=${dependency_file}"
-      -P "${CMAKE_ROOT}/Modules/dependencies.cmake" # TODO hard coded path
+      -P "${CMAKE_ROOT}/Modules/DLangDependencies.cmake" # TODO hard coded path
     )
 
     # load dependencies from file
@@ -55,7 +55,7 @@ macro(add_executable_with_dependencies _target)
       -D "include_directories:STRING=${INCLUDES}"
       -D "source_file:STRING=${source_file}"
       -D "dependency_file:STRING=${dependency_file}"
-      -P "${CMAKE_ROOT}/Modules/dependencies.cmake" # TODO hard coded path
+      -P "${CMAKE_ROOT}/Modules/DLangDependencies.cmake" # TODO hard coded path
       COMMENT "Scanning for dependencies")
   endforeach()
 
