diff --git a/embuilder.py b/embuilder.py
index 6e8d0d9..82c351b 100755
--- a/embuilder.py
+++ b/embuilder.py
@@ -194,7 +194,7 @@ def main():
       library = SYSTEM_LIBRARIES[what]
       library.get_path()
     elif what == 'struct_info':
-      build(C_BARE, ['generated_struct_info.json'])
+      build(C_BARE, ['generated_struct_info.json'], ['-std=gnu17'])
     elif what == 'native_optimizer':
       build(C_BARE, ['optimizer.2.exe'], ['-O2', '-s', 'WASM=0'])
     elif what == 'icu':
diff --git a/tools/gen_struct_info.py b/tools/gen_struct_info.py
index 29ad775..1e17433 100755
--- a/tools/gen_struct_info.py
+++ b/tools/gen_struct_info.py
@@ -512,7 +512,7 @@ def main(args):
   QUIET = args.quiet
 
   # Avoid parsing problems due to gcc specifc syntax.
-  cpp_opts = ['-D_GNU_SOURCE']
+  cpp_opts = ['-D_GNU_SOURCE', '-std=gnu17']
 
   # Add the user options to the list as well.
   for path in args.includes:
