diff --git a/src/json2msgpack.cpp b/src/json2msgpack.cpp
index 25e12bd..8bca97e 100644
--- a/src/json2msgpack.cpp
+++ b/src/json2msgpack.cpp
@@ -27,6 +27,14 @@
 #include <errno.h>
 #include "rapidjson/document.h"
 
+#ifndef VERSION
+#define VERSION "26d5bb58e37a2ae27f375c1814bc03984c997103"
+#endif
+
+#ifndef LIBB64_VERSION
+#define LIBB64_VERSION "1.2.1"
+#endif
+
 using namespace rapidjson;
 
 typedef struct options_t {
diff --git a/src/msgpack2json.cpp b/src/msgpack2json.cpp
index 5296cde..0f4e279 100644
--- a/src/msgpack2json.cpp
+++ b/src/msgpack2json.cpp
@@ -36,6 +36,14 @@
 #define HEX_PREFIX_BYTE_COUNT 8
 #define BIN_EXT_DESCRIPTION_LENGTH 64
 
+#ifndef VERSION
+#define VERSION "26d5bb58e37a2ae27f375c1814bc03984c997103"
+#endif
+
+#ifndef LIBB64_VERSION
+#define LIBB64_VERSION "1.2.1"
+#endif
+
 using namespace rapidjson;
 
 typedef enum continuous_mode_t {
