diff --git a/src/support/debug.h b/src/support/debug.h
index edb19eb..41ee200 100644
--- a/src/support/debug.h
+++ b/src/support/debug.h
@@ -44,11 +44,6 @@ void setDebugEnabled(const char* types);
 
 #else
 
-// We have an option to build with assertions disabled
-// BYN_ASSERTIONS_ENABLED=OFF, but we currently don't recommend using and we
-// don't test with it.
-#error "binaryen is currently designed to be built with assertions enabled."
-#error "remove these #errors if you want to build without them anyway."
 
 #define BYN_DEBUG_WITH_TYPE(...)                                               \
   do {                                                                         \
