diff --git a/Makefile b/Makefile
index 276e227..5cfc746 100644
--- a/Makefile
+++ b/Makefile
@@ -13,13 +13,13 @@ datadir       = ${datarootdir}
 
 override CXXFLAGS += --std=gnu++11 -g3
 
-all: libprimus_vk.so libnv_vulkan_wrapper.so
+all: libprimus_vk.so32 libnv_vulkan_wrapper.so32
 
-libprimus_vk.so: primus_vk.cpp
-	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -I/usr/include/vulkan -shared -fPIC $^ -o $@ -Wl,-soname,libprimus_vk.so.1 -ldl -lpthread $(LDFLAGS)
+libprimus_vk.so32: primus_vk.cpp
+	$(CXX) -m32 $(CPPFLAGS) $(CXXFLAGS) -I/usr/include/vulkan -shared -fPIC $^ -o $@ -Wl,-soname,libprimus_vk.so.1 -ldl -lpthread $(LDFLAGS)
 
-libnv_vulkan_wrapper.so: nv_vulkan_wrapper.cpp
-	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -I/usr/include/vulkan -shared -fPIC $^ -o $@ -Wl,-soname,libnv_vulkan_wrapper.so.1 -ldl $(LDFLAGS)
+libnv_vulkan_wrapper.so32: nv_vulkan_wrapper.cpp
+	$(CXX) -m32 $(CPPFLAGS) $(CXXFLAGS) -I/usr/include/vulkan -shared -fPIC $^ -o $@ -Wl,-soname,libnv_vulkan_wrapper.so.1 -ldl $(LDFLAGS)
 
 primus_vk_forwarding.h:
 	xsltproc surface_forwarding_functions.xslt /usr/share/vulkan/registry/vk.xml | tail -n +2 > $@
diff --git a/nv_vulkan_wrapper.cpp b/nv_vulkan_wrapper.cpp
index 6194974..f230eea 100644
--- a/nv_vulkan_wrapper.cpp
+++ b/nv_vulkan_wrapper.cpp
@@ -8,7 +8,7 @@ extern "C" VKAPI_ATTR VkResult VKAPI_CALL vk_icdNegotiateLoaderICDInterfaceVersi
 
 
 #ifndef NV_DRIVER_PATH
-#define NV_DRIVER_PATH "/usr/lib64/opengl/nvidia/lib/libGLX_nvidia.so.0"
+#define NV_DRIVER_PATH "/usr/lib/opengl/nvidia/lib/libGLX_nvidia.so.0"
 #endif
 
 class StaticInitialize {
