diff --git a/scripts/trylink b/scripts/trylink
index 6b74f09..fb3c28b 100755
--- a/scripts/trylink
+++ b/scripts/trylink
@@ -57,18 +57,7 @@ check_cc() {
 }
 
 check_libc_is_glibc() {
-    tempname="$(mktemp tmp.XXXXXXXXXX)"
-    echo "\
-	#include <stdlib.h>
-	/* Apparently uclibc defines __GLIBC__ (compat trick?). Oh well. */
-	#if defined(__GLIBC__) && !defined(__UCLIBC__)
-	syntax error here
-	#endif
-	" >"$tempname".c
-    ! $CC $CFLAGS "$tempname".c -c -o "$tempname".o >/dev/null 2>&1
-    exitcode=$?
-    rm -f "$tempname" "$tempname".c "$tempname".o
-    return $exitcode
+    return 1
 }
 
 EXE="$1"
