diff --git a/src/config.cpp b/src/config.cpp
index ebfc2ff..01d215d 100644
--- a/src/config.cpp
+++ b/src/config.cpp
@@ -34,36 +34,11 @@ void enumerate_config_files(std::vector<std::string>& paths)
 {
     static const char *mangohud_dir = "/MangoHud/";
 
-    std::string env_data = get_data_dir();
     std::string env_config = get_config_dir();
 
     if (!env_config.empty())
         paths.push_back(env_config + mangohud_dir + "MangoHud.conf");
-#ifdef _WIN32
-    paths.push_back("C:\\MangoHud.conf");
-#endif
-    std::string exe_path = get_exe_path();
-    auto n = exe_path.find_last_of('/');
-    if (!exe_path.empty() && n != std::string::npos && n < exe_path.size() - 1) {
-        // as executable's name
-        std::string basename = exe_path.substr(n + 1);
-        program_name = basename;
-        if (!env_config.empty())
-            paths.push_back(env_config + mangohud_dir + basename + ".conf");
 
-        // in executable's folder though not much sense in /usr/bin/
-        paths.push_back(exe_path.substr(0, n) + "/MangoHud.conf");
-
-        // find executable's path when run in Wine
-        if (!env_config.empty() && (basename == "wine-preloader" || basename == "wine64-preloader")) {
-            
-            std::string name;
-            if (get_wine_exe_name(name)) {
-                paths.push_back(env_config + mangohud_dir + "wine-" + name + ".conf");
-            }
-            program_name = name;
-        }
-    }
     if (program_name.empty())
         program_name = "Unknown";
 }
