diff --git a/src/config.cpp b/src/config.cpp
index 291fddf..58b8ce9 100644
--- a/src/config.cpp
+++ b/src/config.cpp
@@ -64,31 +64,14 @@ std::string get_program_name() {
 static void enumerate_config_files(std::vector<std::string>& paths) {
     static const char *mangohud_dir = "/MangoHud/";
 
-    const std::string data_dir = get_data_dir();
     const std::string config_dir = get_config_dir();
 
-    const std::string program_name = get_program_name();
-
     if (config_dir.empty()) {
         // If we can't find 'HOME' just abandon hope.
         return;
     }
 
     paths.push_back(config_dir + mangohud_dir + "MangoHud.conf");
-
-    if (!program_name.empty()) {
-        paths.push_back(config_dir + mangohud_dir + program_name + ".conf");
-    }
-
-    const std::string program_dir = get_program_dir();
-    if (!program_dir.empty()) {
-        paths.push_back(program_dir + "/MangoHud.conf");
-    }
-
-    const std::string wine_program_name = get_wine_exe_name();
-    if (!wine_program_name.empty()) {
-        paths.push_back(config_dir + mangohud_dir + "wine-" + wine_program_name + ".conf");
-     }
 }
 
 bool parseConfigFile(overlay_params& params) {
