diff --git a/CMakeLists.txt b/CMakeLists.txt
index 87ee5d9..4f45451 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,6 @@ include(ECMQtDeclareLoggingCategory)
 # https://doc.qt.io/qt-6/portingguide.html
 add_compile_definitions(QT_DISABLE_DEPRECATED_BEFORE=0x050F00)
 
-add_subdirectory (src/libdbusmenuqt)
 add_subdirectory (src)
 
 feature_summary(WHAT ALL)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1a99581..9107ff5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -108,7 +108,6 @@ add_library (materialdecoration MODULE
 
 target_link_libraries (materialdecoration
     PUBLIC
-        dbusmenuqt
         Qt${QT_VERSION_MAJOR}::Core
         Qt${QT_VERSION_MAJOR}::Gui
         # Qt${QT_VERSION_MAJOR}::X11Extras
