--- libcxx-8.0.1.ebuild
+++ libcxx-8.0.1.ebuild
@@ -80,6 +80,8 @@
 }
 
 src_configure() {
+    echo "USING CUSTOM PATCHED VERSION"
+
 	# note: we need to do this before multilib kicks in since it will
 	# alter the CHOST
 	local cxxabi cxxabi_incs
@@ -133,6 +135,7 @@
 		-DLIBCXX_ENABLE_STATIC=$(usex static-libs)
 		-DLIBCXX_CXX_ABI=${cxxabi}
 		-DLIBCXX_CXX_ABI_INCLUDE_PATHS=${cxxabi_incs}
+		-DLIBCXX_USE_COMPILER_RT=ON
 		# we're using our own mechanism for generating linker scripts
 		-DLIBCXX_ENABLE_ABI_LINKER_SCRIPT=OFF
 		-DLIBCXX_HAS_MUSL_LIBC=$(usex elibc_musl)
