diff --git a/src/datastructs-private.h b/src/datastructs-private.h
index 7543341..b0f2cbd 100644
--- a/src/datastructs-private.h
+++ b/src/datastructs-private.h
@@ -30,8 +30,8 @@
 
 struct _WindowData
 {
-	uint window_id;
-	ulong wayland_window_id;
+	unsigned int window_id;
+	unsigned long wayland_window_id;
 	GMenu *menu_model;
 	guint menu_model_export_id;
 	GSList *menus;
diff --git a/src/support.c b/src/support.c
index fdad333..26b6b81 100644
--- a/src/support.c
+++ b/src/support.c
@@ -121,7 +121,7 @@ G_GNUC_INTERNAL void gtk_widget_disconnect_settings(GtkWidget *widget)
 }
 
 #if (GTK_MAJOR_VERSION < 3) || defined(GDK_WINDOWING_WAYLAND)
-static uint watcher_id = 0;
+static unsigned int watcher_id = 0;
 
 static gboolean is_dbus_present()
 {
