diff --git a/tools/system_libs.py b/tools/system_libs.py
index 678a2f7..70d48ea 100755
--- a/tools/system_libs.py
+++ b/tools/system_libs.py
@@ -216,7 +216,7 @@ class Library(object):
 
   # A list of flags to pass to emcc.
   # The flags for the parent class is automatically inherited.
-  cflags = ['-Werror']
+  cflags = ['-Werror', '-std=gnu17']
 
   # A list of directories to put in the include path when building.
   # This is a list of tuples of path components.
