diff --git a/src/plugins/deditor/dcompletionassist.cpp b/src/plugins/deditor/dcompletionassist.cpp
index 33a5e45..31ecc28 100644
--- a/src/plugins/deditor/dcompletionassist.cpp
+++ b/src/plugins/deditor/dcompletionassist.cpp
@@ -99,7 +99,7 @@ DCompletionAssistInterface::DCompletionAssistInterface(QTextDocument *textDocume
 																																																							int position,
 																																																							const QString &fileName,
 																																																							TextEditor::AssistReason reason)
-	: AssistInterface(textDocument, position, fileName, reason)
+	: AssistInterface(textDocument, position, Utils::FilePath::fromFileInfo(QFileInfo(fileName)), reason)
 {
 }
 // --------------------------------------------------------------------------------
