--- compiler-rt-8.0.1.ebuild
+++ compiler-rt-8.0.1.ebuild
@@ -55,6 +55,8 @@
 }
 
 src_configure() {
+    echo "USING CUSTOM PATCHED VERSION"
+
 	# pre-set since we need to pass it to cmake
 	BUILD_DIR=${WORKDIR}/${P}_build
 
@@ -80,6 +82,7 @@
 		-DCOMPILER_RT_BUILD_PROFILE=OFF
 		-DCOMPILER_RT_BUILD_SANITIZERS=OFF
 		-DCOMPILER_RT_BUILD_XRAY=OFF
+		-DCOMPILER_RT_USE_BUILTINS_LIBRARY=ON
 	)
 
 	if use prefix && [[ "${CHOST}" == *-darwin* ]] ; then
