diff --git a/include/exceptions.hpp b/include/exceptions.hpp
index 8f9f865..7ac9936 100644
--- a/include/exceptions.hpp
+++ b/include/exceptions.hpp
@@ -2,6 +2,7 @@
 #include <fmt.hpp>
 #include <stdexcept>
 #include <utils.hpp>
+#include <string>
 
 namespace suex {
 
diff --git a/include/fmt.hpp b/include/fmt.hpp
index 57250d6..76be841 100644
--- a/include/fmt.hpp
+++ b/include/fmt.hpp
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <memory>
+#include <string>
 
 // https://stackoverflow.com/a/26221725/4579708
 template <typename... Args>
diff --git a/src/file.cpp b/src/file.cpp
index 34fa3ac..622b34b 100644
--- a/src/file.cpp
+++ b/src/file.cpp
@@ -3,6 +3,7 @@
 #include <file.hpp>
 #include <logger.hpp>
 #include <sstream>
+#include <string>
 
 off_t file::File::Size() const { return Status().st_size; }
 
@@ -138,7 +139,7 @@ void file::File::ReadLine(
       line[read - 1] = '\0';
     }
 
-    callback(line_t{.lineno = lineno, .txt = line});
+    callback(line_t{line, lineno});
   }
 }
 file::File::~File() {
diff --git a/src/path.cpp b/src/path.cpp
index c4b8e3a..ff86d61 100644
--- a/src/path.cpp
+++ b/src/path.cpp
@@ -4,6 +4,7 @@
 #include <gsl/gsl>
 #include <logger.hpp>
 #include <sstream>
+#include <string>
 
 const std::string path::Locate(const std::string &path, bool searchInPath) {
   if (path.empty()) {
diff --git a/src/perm.cpp b/src/perm.cpp
index d6fcbbb..ca4bf22 100644
--- a/src/perm.cpp
+++ b/src/perm.cpp
@@ -1,6 +1,7 @@
 #include <exceptions.hpp>
 #include <logger.hpp>
 #include <sstream>
+#include <string>
 
 using suex::permissions::Entity;
 using suex::permissions::Group;
diff --git a/src/utils.cpp b/src/utils.cpp
index 134d56d..3b9f2f0 100644
--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -2,6 +2,8 @@
 #include <gsl/gsl>
 #include <iomanip>
 #include <logger.hpp>
+#include <string>
+#include <sstream>
 
 using suex::permissions::User;
 
