--- llvm-9.0.0.9999.ebuild
+++ llvm-9.0.0.9999.ebuild
@@ -116,12 +116,24 @@
 }
 
 multilib_src_configure() {
+    echo "USING CUSTOM PATCHED VERSION"
+
 	local ffi_cflags ffi_ldflags
 	if use libffi; then
 		ffi_cflags=$($(tc-getPKG_CONFIG) --cflags-only-I libffi)
 		ffi_ldflags=$($(tc-getPKG_CONFIG) --libs-only-L libffi)
 	fi
 
+    if [[ "${CHOST}" == *"x86_64"* ]]; then
+        export CHOST=x86_64-pc-linux
+        TARGET_TRIPLET=x86_64-pc-linux
+    else
+        export CHOST=i686-pc-linux
+        TARGET_TRIPLET=i686-pc-linux
+    fi
+
+    echo ">>> target triplet: ${TARGET_TRIPLET}"
+
 	local libdir=$(get_libdir)
 	local mycmakeargs=(
 		# disable appending VCS revision to the version to improve
@@ -146,7 +158,9 @@
 
 		-DWITH_POLLY=OFF # TODO
 
-		-DLLVM_HOST_TRIPLE="${CHOST}"
+		-DLLVM_TARGET_TRIPLE_ENV="${TARGET_TRIPLET}"
+		-DLLVM_HOST_TRIPLE="${TARGET_TRIPLET}"
+		-DLLVM_DEFAULT_TARGET_TRIPLE="${TARGET_TRIPLET}"
 
 		-DFFI_INCLUDE_DIR="${ffi_cflags#-I}"
 		-DFFI_LIBRARY_DIR="${ffi_ldflags#-L}"
@@ -196,16 +210,6 @@
 		)
 	fi
 
-	if tc-is-cross-compiler; then
-		local tblgen="${EPREFIX}/usr/lib/llvm/${SLOT}/bin/llvm-tblgen"
-		[[ -x "${tblgen}" ]] \
-			|| die "${tblgen} not found or usable"
-		mycmakeargs+=(
-			-DCMAKE_CROSSCOMPILING=ON
-			-DLLVM_TABLEGEN="${tblgen}"
-		)
-	fi
-
 	# workaround BMI bug in gcc-7 (fixed in 7.4)
 	# https://bugs.gentoo.org/649880
 	# apply only to x86, https://bugs.gentoo.org/650506
@@ -222,6 +226,16 @@
 }
 
 multilib_src_compile() {
+    if [[ "${CHOST}" == *"x86_64"* ]]; then
+        export CHOST=x86_64-pc-linux
+        TARGET_TRIPLET=x86_64-pc-linux
+    else
+        export CHOST=i686-pc-linux
+        TARGET_TRIPLET=i686-pc-linux
+    fi
+
+    echo ">>> target triplet: ${TARGET_TRIPLET}"
+
 	cmake-utils_src_compile
 
 	pax-mark m "${BUILD_DIR}"/bin/llvm-rtdyld
@@ -242,6 +256,16 @@
 }
 
 src_install() {
+    if [[ "${CHOST}" == *"x86_64"* ]]; then
+        export CHOST=x86_64-pc-linux
+        TARGET_TRIPLET=x86_64-pc-linux
+    else
+        export CHOST=i686-pc-linux
+        TARGET_TRIPLET=i686-pc-linux
+    fi
+
+    echo ">>> target triplet: ${TARGET_TRIPLET}"
+
 	local MULTILIB_CHOST_TOOLS=(
 		/usr/lib/llvm/${SLOT}/bin/llvm-config
 	)
@@ -258,6 +282,16 @@
 }
 
 multilib_src_install() {
+    if [[ "${CHOST}" == *"x86_64"* ]]; then
+        export CHOST=x86_64-pc-linux
+        TARGET_TRIPLET=x86_64-pc-linux
+    else
+        export CHOST=i686-pc-linux
+        TARGET_TRIPLET=i686-pc-linux
+    fi
+
+    echo ">>> target triplet: ${TARGET_TRIPLET}"
+
 	cmake-utils_src_install
 
 	# move headers to /usr/include for wrapping
@@ -268,6 +302,16 @@
 }
 
 multilib_src_install_all() {
+    if [[ "${CHOST}" == *"x86_64"* ]]; then
+        export CHOST=x86_64-pc-linux
+        TARGET_TRIPLET=x86_64-pc-linux
+    else
+        export CHOST=i686-pc-linux
+        TARGET_TRIPLET=i686-pc-linux
+    fi
+
+    echo ">>> target triplet: ${TARGET_TRIPLET}"
+
 	local revord=$(( 9999 - ${SLOT} ))
 	newenvd - "10llvm-${revord}" <<-_EOF_
 		PATH="${EPREFIX}/usr/lib/llvm/${SLOT}/bin"
