From 3427accd188e9c85c08a25b84615098175e4f49b Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgads@gmail.com>
Date: Mon, 23 Dec 2019 11:39:39 +0200
Subject: [PATCH] Adapt to 4.12

---
 Parsers/CppParser/cppdocumentparser.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Parsers/CppParser/cppdocumentparser.cpp b/Parsers/CppParser/cppdocumentparser.cpp
index d543ef5..091588b 100644
--- a/Parsers/CppParser/cppdocumentparser.cpp
+++ b/Parsers/CppParser/cppdocumentparser.cpp
@@ -531,7 +531,7 @@ void CppDocumentParser::reparseProject()
     return;
   }
 
-  const Utils::FilePathList projectFiles = d->activeProject->files( ProjectExplorer::Project::SourceFiles );
+  const Utils::FilePaths projectFiles = d->activeProject->files( ProjectExplorer::Project::SourceFiles );
   const QStringList fileList             = Utils::transform( projectFiles, &Utils::FilePath::toString );
 
   const QStringSet fileSet = d->getCppFiles( fileList.toSet() );
