diff --git a/plasmoid/metadata.desktop b/plasmoid/metadata.desktop
index 48ed0ef..64e231a 100644
--- a/plasmoid/metadata.desktop
+++ b/plasmoid/metadata.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Name=Media Player +
+Name=Media Player Plus
 Name[ar]=مشغّل وسائط
 Name[bs]=Izvođač medija
 Name[ca]=Reproductor multimèdia
@@ -11,18 +11,19 @@ Name[el]=Αναπαραγωγέας πολυμέσων
 Name[en_GB]=Media Player
 Name[es]=Reproductor multimedia
 Name[et]=Meediamängija
-Name[eu]=Multimedia-jotzailea
+Name[eu]=Euskarri jotzailea
 Name[fi]=Mediasoitin
 Name[fr]=Lecteur multimédia
 Name[gl]=Reprodutor multimedia
 Name[he]=נגן מדיה
 Name[hu]=Médialejátszó
 Name[ia]=Media Player (Reproductor de Media)
+Name[id]=Pemutar Media
 Name[is]=Margmiðlunarspilari
 Name[it]=Lettore multimediale
 Name[ja]=メディアプレーヤー
 Name[ko]=미디어 재생기
-Name[lt]=Kūrinių leistuvė
+Name[lt]=Medijos leistuvė
 Name[nb]=Mediespiller
 Name[nds]=Medienafspeler
 Name[nl]=Mediaspeler
@@ -39,6 +40,7 @@ Name[sr@ijekavian]=медија плејер
 Name[sr@ijekavianlatin]=medija plejer
 Name[sr@latin]=medija plejer
 Name[sv]=Mediaspelare
+Name[tg]=Плеери медиа
 Name[tr]=Ortam Yürütücüsü
 Name[uk]=Програвач
 Name[x-test]=xxMedia Playerxx
@@ -56,19 +58,19 @@ Comment[el]=Χειριστήρια αναπαραγωγέα πολυμέσων
 Comment[en_GB]=Media Player Controls
 Comment[es]=Controles del reproductor multimedia
 Comment[et]=Meediamängija juhtelemendid
-Comment[eu]=Multimedia-jotzailearen kontrolak
+Comment[eu]=Euskarri jotzaile kontrolak
 Comment[fi]=Mediasoittimen säätimet
 Comment[fr]=Contrôles du lecteur multimédia
 Comment[gl]=Controis do reprodutor multimedia
 Comment[he]=פקדי ניגון מדיה
 Comment[hu]=Médialejátszó vezérlők
 Comment[ia]=Controlos de Reproductor de Multimedia
-Comment[id]=Kendali Player Media
+Comment[id]=Kendali Pemutar Media
 Comment[is]=Stjórntæki margmiðlunarspilara
 Comment[it]=Controlli del lettore multimediale
 Comment[ja]=メディアプレーヤーコントロール
 Comment[ko]=미디어 재생기 제어
-Comment[lt]=Kūrinių leistuvės valdikliai
+Comment[lt]=Medijos leistuvės valdikliai
 Comment[nb]=Styring for mediespiller
 Comment[nds]=Medienafspeler stüern
 Comment[nl]=Besturing van mediaspeler
